OBS-Studio doesn't compile with GCC 13

### Operating System Info

Other

### Other OS

Arch Linux

### OBS Studio Version

29.1.0

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

None

### OBS Studio Crash Log URL

None

### Expected Behavior

OBS Studio should build normally with the provided Source Code (and a pkgbuild from the AUR) as it did before the update from GCC 12 > 13.

Didn't touched the /etc/makepkg.conf and all Compiler-Flags are unmodified...

### Current Behavior

Instead with GCC 13 i'll get those Error Messages and the build process stopped

+++

/home/gnatz/.cache/paru/clone/obs-studio-tytan652/src/obs-studio/plugins/obs-ffmpeg/ffmpeg-mux/ffmpeg-mux.c: In function ‘ffmpeg_mux_io_thread’:
/home/gnatz/.cache/paru/clone/obs-studio-tytan652/src/obs-studio/plugins/obs-ffmpeg/ffmpeg-mux/ffmpeg-mux.c:826:55: error: ‘next_seek_position’ may be used uninitialized [-Werror=maybe-uninitialized]
  826 |                                 current_seek_position =
      |                                 ~~~~~~~~~~~~~~~~~~~~~~^
  827 |                                         next_seek_position + chunk_used;
      |                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gnatz/.cache/paru/clone/obs-studio-tytan652/src/obs-studio/plugins/obs-ffmpeg/ffmpeg-mux/ffmpeg-mux.c:730:18: note: ‘next_seek_position’ was declared here
  730 |         uint64_t next_seek_position;
      |                  ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [plugins/obs-ffmpeg/ffmpeg-mux/CMakeFiles/obs-ffmpeg-mux.dir/build.make:76: plugins/obs-ffmpeg/ffmpeg-mux/CMakeFiles/obs-ffmpeg-mux.dir/ffmpeg-mux.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1718: plugins/obs-ffmpeg/ffmpeg-mux/CMakeFiles/obs-ffmpeg-mux.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
==> ERROR: A failure occurred in build().

**--- AND ---**

In function 'DecodeTEA',
    inlined from 'HandleInvoke' at /home/gnatz/.cache/paru/clone/obs-studio-tytan652/src/obs-studio/plugins/obs-outputs/librtmp/rtmp.c:3168:21:
/home/gnatz/.cache/paru/clone/obs-studio-tytan652/src/obs-studio/plugins/obs-outputs/librtmp/rtmp.c:4592:7: error: '*(uint32_t *)out' may be used uninitialized [-Werror=maybe-uninitialized]
4592 |     y = v[0];
    |     ~~^~~~~~
cc1: all warnings being treated as errors
make[2]: *** [plugins/obs-outputs/CMakeFiles/obs-outputs.dir/build.make:272: plugins/obs-outputs/CMakeFiles/obs-outputs.dir/librtmp/rtmp.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1744: plugins/obs-outputs/CMakeFiles/obs-outputs.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...

Maybe there are even more Errors but the build process stopped there for me, so i don't know.

### Steps to Reproduce

1. Download and extract the source code
2. Compile
3. Wait for Error...
...


### Anything else we should know?

_No response_

--- a/cmake/Modules/CompilerConfig.cmake
+++ b/cmake/Modules/CompilerConfig.cmake
@@ -100,6 +100,13 @@ else()
   endif()
 
   option(CALM_DEPRECATION "Keep deprecated-declarations as warnings" OFF)
+  #[[
+    Note about -Wmaybe-uninitialized on GCC, this warning seems to be subject of various regressions and false positives. This
+    warning is set to not turn into an error.
+
+    - https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105562 for 12.1.0
+    - https://github.com/obsproject/obs-studio/issues/8850 for 13.1.1
+  ]]
   add_compile_options(
     -Werror
     -Wextra
@@ -117,19 +124,13 @@ else()
     "$<$<CONFIG:DEBUG>:-DDEBUG=1;-D_DEBUG=1>"
     "$<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-Wnull-conversion;-fcolor-diagnostics;-Wno-error=shorten-64-to-32>"
     "$<$<COMPILE_LANG_AND_ID:C,AppleClang,Clang>:-Wnull-conversion;-fcolor-diagnostics;-Wno-error=shorten-64-to-32>"
-    "$<$<COMPILE_LANG_AND_ID:CXX,GNU>:-Wconversion-null>"
+    "$<$<COMPILE_LANG_AND_ID:CXX,GNU>:-Wconversion-null;-Wno-error=maybe-uninitialized>"
+    "$<$<COMPILE_LANG_AND_ID:C,GNU>:-Wno-error=maybe-uninitialized>"
     "$<$<BOOL:${CALM_DEPRECATION}>:-Wno-error=deprecated-declarations>")
 
-  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    # GCC on aarch64 emits type-limits warnings that do not appear on x86_64
-    if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
-      add_compile_options(-Wno-error=type-limits)
-    endif()
-
-    # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105562
-    if(CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "12.1.0")
-      add_compile_options(-Wno-error=maybe-uninitialized)
-    endif()
+  # GCC on aarch64 emits type-limits warnings that do not appear on x86_64
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+    add_compile_options(-Wno-error=type-limits)
   endif()
 
   if(OBS_CODESIGN_LINKER)

